pallets/stream/src/lib.rs

```diff
@@ -267,12 +281,14 @@ pub mod pallet {
 				StreamEntryOf::<T> {
 					digest: stream_digest,
 					creator: creator.clone(),
-					schema: schema_id,
+					schema: schema_id.clone(),
 					registry: registry_id,
 					revoked: false,
 				},
 			);
 
+			<UniqueStreams<T>>::insert(&identifier, stream_digest, false);
```

Comment:
+1. this can become the revocation registry as well. We can think of removing the revoke field from the StreamEntry and make this structure the state proof.

Reply:
True. Just that we have to make it clear, (or do 2 queries from SDK) that one is state check, and another is identifier latest details check.

Reply:
moved 'creator' and 'revoked_by' also into the structure.